src/shogun/mathematics/Statistics.cpp

```diff
@@ -325,6 +325,7 @@ SGVector<int32_t> CStatistics::sample_indices(int32_t sample_size, int32_t N)
 	int32_t* idxs=SG_MALLOC(int32_t,N);
 	int32_t i, rnd;
 	int32_t* permuted_idxs=SG_MALLOC(int32_t,sample_size);
+	auto rng = std::unique_ptr<CRandom>(new CRandom(get_global_seed()));
```

Comment:
no need for the `get_global_seed` argument as the default ctor uses global seed anyways right?